pkl/evaluator_options.go

```diff
@@ -506,6 +522,12 @@ var WithExternalResourceReader = func(scheme string, spec ExternalReader) func(o
 	}
 }
 
+var WithTraceMode = func(mode TraceMode) func(opts *EvaluatorOptions) {
+	return func(opts *EvaluatorOptions) {
+		opts.TraceMode = mode
+	}
+}
```

Comment:
We generally don't provide `WithXXX` method for simple setters. Users can set `opts.TraceMode` directly themselves by passing in their own function:

```go
pkl.NewEvaluator(context.Background(), pkl.PreconfiguredOptions, func(opts *EvaluatorOptions) {
  opts.TraceMode = pkl.TraceCompact
})
```